src/components/wizard/wizard.component.ts

```diff
@@ -347,6 +349,28 @@ export class WizardComponent implements AfterViewInit, OnDestroy {
         return this.steps.toArray()[index];
     }
 
+    /**
+     * If a step in the wizard becomes invalid, all steps sequentially after
+     * it, should become unvisited and incomplete
+     */
+    setNextStepsUnvisited(currentStep: WizardStepComponent, cb?: (step: WizardStepComponent | MarqueeWizardStepComponent) => void): void {
+        const steps = this.steps.toArray();
+        const affected = steps.slice(this.step + 1);
+
+        affected.forEach(step => {
+
+            // if the step is not the current step then also mark it as unvisited
+            if (step !== currentStep) {
+                step.visited = false;
```

Comment:
Is this ever going to emit visitedChange? Would we need to use the `setVisited` function here instead? As this is the component changing the visited state and not the user we would need to ensure that we emit the new value.